src/components/Tooltip/index.js

```diff
@@ -0,0 +1,202 @@
+  direction,
+  isOpen,
+  showOnHover,
+  tooltipProps: tooltipOptsProp,
+  tooltipWidth,
+  ...rest
+}) => {
+  const [tooltipId] = useState(
+    `tooltip-${Date.now() * Math.floor(100 * Math.random())}`
+  );
+  const tooltipRef = useRef(null);
+
+  const [hovered, setHovered] = useState(false);
+  const [tooltipHeight, setTooltipHeight] = useState(0);
+
+  const closeFunc = useCallback(closeable || function() {}, []);
+
+  const handleSetHover = value => () => {
+    if (showOnHover) {
+      setHovered(value);
+    }
+  };
+
+  const handleEscape = e => {
+    if (closeable && e.key === 'Escape') {
+      closeable();
+    }
+  };
+
+  useEffect(() => {
+    if (closeOnEscape) {
+      document.addEventListener('keydown', handleEscape);
+
+      return () => {
+        document.removeEventListener('keydown', handleEscape);
+      };
+    }
+
+    return () => {};
+  }, []);
+
+  useEffect(() => {
+    if (tooltipRef.current) {
+      setTooltipHeight(tooltipRef.current.offsetHeight);
+    }
+  }, []); // empty deps ensure this only gets called once
+
+  const {
+    style: tooltipStyle,
+    className: tooltipClassName,
+    ...tooltipOpts
+  } = tooltipOptsProp;
+
+  const tooltipClassNames = classNames(
+    styles.Tooltip,
+    styles[direction],
+    tooltipClassName,
+    {
+      [styles.open]: isOpen || hovered,
+    }
+  );
+
+  const offsets = useMemo(() => {
+    // Properly center tooltips on the left and right to the center
+    // of the actual element we are attaching the tooltip.
+    if (direction === 'left' || direction === 'right') {
+      return {
+        marginTop: -(tooltipHeight / 2),
+      };
+    }
+
+    return {};
+  });
+
+  const isText = typeof tooltipContent === 'string';
+
+  const tooltipWrapperProps = {
+    style: {
+      width: (tooltipStyle && tooltipStyle.width) || tooltipWidth,
+      ...offsets,
+    },
+    className: tooltipClassNames,
+  };
+
+  const TooltipContent = tooltipContent && (
+    <div
+      id={tooltipId}
+      aria-hidden={isOpen}
+      ref={tooltipRef}
+      {...tooltipWrapperProps}
+    >
+      <div
+        style={tooltipStyle}
+        className={styles.tooltipInnerWrapper}
+        {...tooltipOpts}
+      >
+        {!!closeable && (
+          <button className={styles.tooltipClose} onClick={closeFunc}>
+            <Icon name="close" />
+          </button>
+        )}
+
+        {isText ? (
+          <div className={styles.textContent}>{tooltipContent}</div>
+        ) : (
+          tooltipContent
+        )}
+      </div>
+    </div>
+  );
+
+  return (
+    <div className={styles.wrapper} {...rest}>
+      {TooltipContent}
+      <div
+        aria-describedby={tooltipId}
+        onMouseEnter={handleSetHover(true)}
+        onMouseLeave={handleSetHover(false)}
+        className={styles.original}
+      >
+        {children}
+      </div>
+    </div>
+  );
+};
+
+Tooltip.propTypes = {
+  children: PropTypes.node.isRequired,
+  closeable: PropTypes.func,
```

Comment:
How about the naming convention `onClose`?

Reply:
Ah, right. I had that before. Thanks for reminding me of switching it back